Src/java/quick/src/main/resources/org/hl7/fhir/FHIRHelpers-4.0.1.cql

```diff
@@ -11,16 +11,70 @@ define function ToInterval(period FHIR.Period):
         else
             Interval[period."start".value, period."end".value]
 
+define function ToCalendarUnit(unit System.String):
+    case unit
+        when 'ms' then 'millisecond'
+        when 's' then 'second'
+        when 'min' then 'minute'
+        when 'h' then 'hour'
+        when 'd' then 'day'
+        when 'wk' then 'week'
+        when 'mo' then 'month'
+        when 'a' then 'year'
+        else unit
+    end
+
 define function ToQuantity(quantity FHIR.Quantity):
     case
         when quantity is null then null
         when quantity.value is null then null
-        when quantity.system is null or quantity.system.value = 'http://unitsofmeasure.org' then
-            System.Quantity { value: quantity.value.value, unit: quantity.code.value }
+        when quantity.comparator is not null then
+            Message(null, true, 'FHIRHelpers.ToQuantity.ComparatorQuantityNotSupported', 'Error', 'FHIR Quantity value has a comparator and cannot be converted to a System.Quantity value.')
+        when quantity.system is null or quantity.system.value = 'http://unitsofmeasure.org'
+              or quantity.system.value = 'http://hl7.org/fhirpath/CodeSystem/calendar-units' then
+            System.Quantity { value: quantity.value.value, unit: ToCalendarUnit(Coalesce(quantity.code.value, quantity.unit.value, '1')) }
         else
-            Message(null, true, 'FHIRHelpers.ToQuantity.InvalidFHIRQuantity', 'Error', 'Invalid FHIR Quantity code: ' & quantity.code.value)
+            Message(null, true, 'FHIRHelpers.ToQuantity.InvalidFHIRQuantity', 'Error', 'Invalid FHIR Quantity code: ' & quantity.unit.value & ' (' & quantity.system.value & '|' & quantity.code.value & ')')
     end
 
+define function ToQuantityIgnoringComparator(quantity FHIR.Quantity):
```

Comment:
Given this function, can't `ToQuantity` (above) be simplified further to:
```
define function ToQuantity(quantity FHIR.Quantity):
  if quantity.comparator is not null then
    Message(null, true, 'FHIRHelpers.ToQuantity.ComparatorQuantityNotSupported', 'Error', 'FHIR Quantity value has a comparator and cannot be converted to a System.Quantity value.')
  else
    ToQuantityIgnoringComparator(quantity)
```

Reply:
It definitely could be, but I was worried about introducing another call so I opted against it, given this is low-level conversion code.